shenyu-plugin/shenyu-plugin-redirect/src/main/java/org/apache/shenyu/plugin/redirect/RedirectPlugin.java

```diff
@@ -86,4 +87,9 @@ protected Mono<Void> doExecute(final ServerWebExchange exchange, final ShenyuPlu
             return response.setComplete();
         }
     }
+
+    @Override
+    protected Mono<Void> handleRuleIfNull(final String pluginName, final ServerWebExchange exchange, final ShenyuPluginChain chain) {
```

Comment:
remove this， if not match，donot do this.

Reply:
like this ? just print log error information ,the request chain is not interrupted, and the error result is returned.

```java
@Override
protected Mono<Void> handleRuleIfNull(final String pluginName, final ServerWebExchange exchange, final ShenyuPluginChain chain) {
    LOG.error("can not match rule data: {}", pluginName);
    return chain.execute(exchange);
}
```